{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Align\n",
    "from Bio import SeqIO\n",
    "from Bio import pairwise2\n",
    "from Bio.pairwise2 import format_alignment\n",
    "import pandas as pd \n",
    "\n",
    "\n",
    "# Function read sequences from a file as a string and initializes \n",
    "# two seperate string variables to compare \n",
    "def import_seq():\n",
    "    global seq1_eagle\n",
    "    seq1_eagle = SeqIO.read(\"gene_nuccore_refseqrna_RHO_GOLDEN_EAGLE.fasta\", \"fasta\")\n",
    "    global seq2_owl \n",
    "    seq2_owl = SeqIO.read(\"gene_nuccore_refseqrna_RHO_BURROWING_OWL.fasta\", \"fasta\")\n",
    "\n",
    "# Function that finds the GC content of a DNA sequence \n",
    "# assuming the seq passed in is without flaws (no extra spaces or error base pairs)\n",
    "def find_gc_content(dna_seq):\n",
    "      g_count = dna_seq.count(\"G\")\n",
    "      c_count = dna_seq.count(\"C\")\n",
    "      answer = ((g_count + c_count)/len(dna_seq)) * 100\n",
    "      \n",
    "      return round(answer, 2)\n",
    "  \n",
    "# TO-DO\n",
    "# Function that initalized the 2D array that will hold the 'scores' for the \n",
    "# Needleman-Wunsch algorithm \n",
    "def init_matrix(seq_1, seq_2): \n",
    "    matrix = []\n",
    "\n",
    "# Function uses Biopython lilbrary  to find alignment between two nucleotide sequences. \n",
    "def find_gaps(seq1, seq2): \n",
    "    alignments = pairwise2.align.globalxx(seq1, seq2)\n",
    "    print(format_alignment(*alignments[0]))\n",
    "    \n",
    "        \n",
    "        \n",
    "# Print basic comparing information about sequences \n",
    "def print_basic_info(seq): \n",
    "    seq_length = len(seq)\n",
    "    gc_content = find_gc_content(seq)\n",
    "    a_count = seq.count(\"A\")\n",
    "    t_count = seq.count(\"T\")\n",
    "    g_count = seq.count(\"G\")\n",
    "    c_count = seq.count(\"C\")\n",
    "    \n",
    "    print(\"Sequence 1:\\n\"+ seq, \"\\n\\n\" +\n",
    "      \"Length:\", str(seq_length) + \"\\n\" + \n",
    "      \"GC Content:\", str(gc_content) + \"%\" + \"\\n\" + \n",
    "      \"A count:\", str(a_count) +\"\\n\" + \n",
    "      \"T count:\", str(t_count) + \"\\n\" +\n",
    "      \"G count:\", str(g_count) + \"\\n\" + \n",
    "      \"C count:\", str(c_count) + \"\\n\")  \n",
    "    \n",
    "    #display_counts(a_count, t_count, g_count, c_count)\n",
    "    \n",
    "#def display_counts(a, t, g, c): \n",
    "    #df = pd.DataFrame(x = \"Nucleotide\", y = \"Occurances\")\n",
    "    #df.plot.bar\n",
    "    #display(df)\n",
    "             \n",
    "def main(): \n",
    "    global seq1_eagle\n",
    "    global seq2_owl\n",
    "    \n",
    "    import_seq() \n",
    "    print_basic_info(seq1_eagle.seq)\n",
    "    print_basic_info(seq2_owl.seq)\n",
    "    \n",
    "    print(\"Markers where the sequences match up:\")\n",
    "    find_gaps(seq1_eagle.seq, seq2_owl.seq)\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    main()\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
